### ISSUE TITLE
Can I use the tensorflow backend?

### ISSUE THREAD

**Issue by yyjjer** [baidu-research/ba-dls-deepspeech#19]
Hi, I want to use tensorflow-1.0 instead of theano as the backend, but get some errors.Could you please help me?

**Comment by srvinay** [comment]
You can use tensorflow, but right now the optimizer and loss function are setup as Theano ops. You will need to replace them with TF ops (`tf.nn.ctc_loss`, `tf.train.MomentumOptimizer`).

**Comment by stan-sack** [comment]
@srvinay what parameters do you pass to `tf.nn.ctc_loss`. Also, how do you apply updates when using `tf.train.MomentumOptimizer`?

I tried using `tf.contrib.keras.optimizers.SGD` but had difficulties converting labels to Sparse Tensors.

Should I be compiling a train function for each batch given that the label length will change and tf.nn.ctc_loss is based on label length?
